### PR TITLE
OCPBUGS-3506: Upgrade Machine Nodes in place

### PR DESCRIPTION
**Pull request by sebsoto**
This PR makes it so that Machine back Windows Nodes will be upgraded in place, in the same way that BYOH Windows nodes are. This fixes an issue relating to having to re-download container images after node upgrade. Machines will now be deleted whenever WMCO has to update the Windows userdata secret.

### CODE REVIEW

**Review by aravindhp — COMMENTED** [review]
Nice work, @sebsoto. Couple of questions that could potential lead to changes.

**Review by saifshaikh48 — COMMENTED** [review]
I want to see if I understand the change to machine deletion properly.
Before, we deleted machines if version annotation didn't match, or there was a private key change
Now we delete machines if there is a userData secret change or a private key change?

File: test/e2e/secrets_test.go
Comment (sebsoto) [reply test/e2e/secrets_test.go:108]: The global context (gc) is only changed when `waitForWindowsNodes` is ran.
This is semi-explained by the below comment:
```
	// waitForWindowsNodes will re-populate gc.machineNodes with the configured nodes found
```

File: test/e2e/secrets_test.go
Comment (sebsoto) [reply test/e2e/secrets_test.go:100]: Yes we should

File: test/e2e/secrets_test.go
Comment (mansikulkarni96) [inline test/e2e/secrets_test.go:103]: is a returns*-> returns

File: test/e2e/secrets_test.go
Comment (mansikulkarni96) [inline test/e2e/secrets_test.go:142]: is this a debug log or should be in the final version of code?

**Review by mansikulkarni96 — CHANGES_REQUESTED** [review]

**Comment by alinaryan** [comment]
Besides Mansi's comments, LGTM

File: controllers/secret_controller.go
Comment (aravindhp) [inline controllers/secret_controller.go:195]: Please make this `updateUserData`

File: test/e2e/create_test.go
Comment (aravindhp) [inline test/e2e/create_test.go:62]: Is there a reason for changing the order of the tests?

Comment (sebsoto) [reply test/e2e/create_test.go:62]: I put this at the end because it is now causing the deletion and creation of a node, so it didn't make sense for me to run some of the creation tests against the original node and some against the new node.

File: test/e2e/secrets_test.go
Comment (sebsoto) [reply test/e2e/secrets_test.go:142]: It should be there, it isn't a new log

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

File: test/e2e/create_test.go
Comment (aravindhp) [reply test/e2e/create_test.go:62]: Please call that out in the commit message.

**Comment by aravindhp** [comment]
/hold

@sebsoto you can remove the hold once this is ready to merge.

**Pull request merged**
